ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/BonsaiWorldStateKeyValueStorage.java

```diff
@@ -79,15 +92,25 @@ public class BonsaiWorldStateKeyValueStorage implements WorldStateStorage, AutoC
 
   protected final Subscribers<BonsaiStorageSubscriber> subscribers = Subscribers.create();
 
+  final BonsaiPreImageProxy preImageProxy;
```

Comment:
I think we could create a subclass of BonsaiWorldStateKeyValueStorage that takes the preImageProxy, since we are only using it to recover hashes for reference test worldstate.  Similarly we could move streamAccounts into the subclass.

Reply:
seems to be good like that